python/pyspark/sql/pandas/conversion.py

```diff
@@ -600,15 +608,39 @@ def _create_from_pandas_with_arrow(
         )
         import pyarrow as pa
 
+        infer_pandas_dict_as_map = (
+            str(self.conf.get("spark.sql.execution.pandas.inferPandasDictAsMap")).lower() == "true"
```

Comment:
I think you should add a test in non-connect test cases, and let Spark Connect reuse the test case if possible.

Reply:
Ah, I added tests into connect test suite to directly compare non-connect and connect behavior like:

```python
self.assertEqual(
    sdf.withColumn("test", SF.col("dict_col")[SF.col("str_col")]).collect(),
    cdf.withColumn("test", CF.col("dict_col")[CF.col("str_col")]).collect(),
)
```

Would it be better to move those tests into non-connect and reuse it?

Reply:
yeah, I think it's better to make the test reused.

Reply:
Sure, let me address it

Reply:
Test moved!